src/mjolnir/graphtilebuilder.cc

```diff
@@ -362,9 +368,24 @@ void GraphTileBuilder::StoreTileData() {
     in_mem.write(reinterpret_cast<const char*>(lane_connectivity_builder_.data()),
                  lane_connectivity_builder_.size() * sizeof(LaneConnectivity));
 
-    // Set the end offset
-    header_builder_.set_end_offset(header_builder_.lane_connectivity_offset() +
-                                   (lane_connectivity_builder_.size() * sizeof(LaneConnectivity)));
+    if (has_osmids_for_nodes()) {
+      if (osmids_for_nodes_.size() != nodes_builder_.size()) {
+        throw std::runtime_error(std::string("length of osmids does not equal length of nodes: ") +
+                                 std::to_string(osmids_for_nodes_.size()) +
+                                 " != " + std::to_string(nodes_builder_.size()));
+      }
+      header_builder_.set_osmids_for_nodes_offset(header_builder_.lane_connectivity_offset() +
```

Comment:
We're missing one line from line 1129 below:

```
header.set_osmids_for_nodes_offset(header.osmids_for_nodes_offset() + shift);
```

Reply:
Thank you!